apps/web/src/ee/billing/components/PlanHeader.tsx

```diff
@@ -36,6 +36,22 @@ export const PlanHeader = () => {
   );
   const [billingInterval, setBillingInterval] = useState<'month' | 'year'>('month');
   const isImprovedBillingEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_IMPROVED_BILLING_ENABLED);
+  const isStripeCheckoutEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_STRIPE_CHECKOUT_ENABLED);
+
+  const checkoutUrl = isStripeCheckoutEnabled ? '/v1/billing/checkout-session' : '/v1/billing/checkout';
+
+  const checkoutOnSuccess = (data) => {
+    if (isStripeCheckoutEnabled) {
+      window.location.href = data.sessionUrl;
```

Comment:
You can make this more explicit

```suggestion
      window.location.href = data.stripeCheckoutUrl;
```